### PR TITLE
feat: button: adds two state button component

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**

### CODE REVIEW

File: src/components/Button/TwoStateButton/TwoStateButton.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            buttonTextClassNames: string,
+            text: string
+        ): JSX.Element => (
+            <span className={buttonTextClassNames}>
+                {text ? text : 'Button'}
+            </span>
+        );
+
+        return (
+            <button
```
Comment (ychhabra-eightfold) [inline src/components/Button/TwoStateButton/TwoStateButton.tsx:150]: Why arent we using BaseButton here ?

File: src/components/Button/button.module.scss [hunk 2]
```diff
@@ -398,6 +448,66 @@
     }
 }
 
+.two-state-button {
```
Comment (ychhabra-eightfold) [inline src/components/Button/button.module.scss:451]: Can we move out two state button into its own scss file, its getting crowded here

File: src/components/Button/Button.types.ts [hunk 3]
```diff
@@ -44,15 +55,36 @@ export interface SplitButtonProps
...
+     * The button counter number.
      */
-    type?: ButtonType;
+    counter?: number;
     /**
-     * Ref of the button
+     * The button icon 1.
      */
-    ref?: Ref<HTMLButtonElement>;
+    iconOne?: IconName;
```
Comment (ychhabra-eightfold) [inline src/components/Button/Button.types.ts:75]: We also have Icon only two state buttons, for eg starring candidate. Will this component handle that, or that will be its own thing ?

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Button/button.module.scss (hunk 2 above)
Comment (dkilgore-eightfold) [reply src/components/Button/button.module.scss:451]: It uses button classes, it's a button. Keeping it here should be necessary.

File: src/components/Button/TwoStateButton/TwoStateButton.tsx (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Button/TwoStateButton/TwoStateButton.tsx:150]: BaseButton is already overloaded, better to keep these separate.

**Review by ychhabra-eightfold — APPROVED** [review]

**Pull request merged**